### PR TITLE
[CDAP-14135] add more trace logging for dataset service and refactor logback-test.xml

### PR DESCRIPTION
**Pull request by anew**
this is to be able to debug the flaky builds.

Also removes all the duplicate ```logback-test.xml``` from the build - instead includes cdap-common:test:test-jar as dependencies in all sub-projects that had their own ```logback-test.xml```.

And updates the BUILD.rst not mention data-quality-app any more.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java [hunk 1]
```diff
@@ -191,15 +191,19 @@ public void setMetadataPublisher(MetadataPublisher metadataPublisher) {
...
       AuthorizationUtil.ensureOnePrivilege(instance, EnumSet.allOf(Action.class),
                                            authorizationEnforcer, authenticationContext.getPrincipal());
+      LOG.info("---- Authorized GET for dataset {}", instance.getDataset());
     }
     // Application Deployment first makes a call to the dataset service to check if the instance already exists with
     // a different type. To make sure that that call responds with the right exceptions if necessary, first fetch the
     // meta from the cache and throw appropriate exceptions if necessary.
     DatasetMeta datasetMeta;
     try {
+      LOG.info("---- Retrieving instance metadata from cache for dataset {}", instance.getDataset());
```
Comment (albertshau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java]: are these intentionally info and not trace?

Comment (anew) [reply cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java]: oops by mistake

File: BUILD.rst [hunk 2]
```diff
@@ -55,13 +55,13 @@ CDAP Sandbox and Distributed CDAP
 - Build CDAP Sandbox distribution ZIP::
 
     MAVEN_OPTS="-Xmx1024m" mvn clean package \
-    -pl cdap-standalone,cdap-app-templates/cdap-etl,cdap-app-templates/cdap-data-quality,cdap-app-templates/cdap-program-report,cdap-examples \
```
Comment (albertshau) [inline BUILD.rst:58]: nice, I was just about to open a PR to fix this.

**Review by albertshau — APPROVED** [review]
lgtm

**Comment by anew** [comment]
build failed due to unrelated flaky tests :( I am merging this now so that we have the improved trace logging in all builds.

**Pull request merged**